nmap.go

```diff
@@ -117,52 +118,107 @@ func (s *Scanner) Run() (*Run, error) {
 }
 
 // RunAsync runs nmap asynchronously and returns error.
-func (s *Scanner) RunAsync() error {
+func (s *Scanner) RunAsync() (<-chan []byte, <-chan []byte, error) {
+	stdoutChannel := make(chan []byte)
+	stderrChannel := make(chan []byte)
+
 	// Enable XML output.
 	s.args = append(s.args, "-oX")
 
 	// Get XML output in stdout instead of writing it in a file.
 	s.args = append(s.args, "-")
 	s.cmd = exec.Command(s.binaryPath, s.args...)
 
+	// Get CMD Stderr Pipe
 	stderr, err := s.cmd.StderrPipe()
 	if err != nil {
-		return fmt.Errorf("unable to get error output from asynchronous nmap run: %v", err)
+		return nil, nil, fmt.Errorf("unable to get error output from asynchronous nmap run: %v", err)
 	}
 
+	// Get CMD Stdout Pipe
 	stdout, err := s.cmd.StdoutPipe()
 	if err != nil {
-		return fmt.Errorf("unable to get standard output from asynchronous nmap run: %v", err)
+		return nil, nil, fmt.Errorf("unable to get standard output from asynchronous nmap run: %v", err)
 	}
 
-	s.stdout = *bufio.NewScanner(stdout)
-	s.stderr = *bufio.NewScanner(stderr)
-
 	if err := s.cmd.Start(); err != nil {
-		return fmt.Errorf("unable to execute asynchronous nmap run: %v", err)
+		return nil, nil, fmt.Errorf("unable to execute asynchronous nmap run: %v", err)
 	}
 
+	// Stream stdout to the stdoutChannel
+	go func() {
+		defer close(stdoutChannel)
+		for {
+			buf := make([]byte, 1024)
+			bytesRead, err := stdout.Read(buf)
+			if err != nil {
+				if err != io.EOF {
+					log.Fatal(err)
+				}
+				if bytesRead == 0 {
+					break
+				}
+			}
+			stdoutChannel <- buf[:bytesRead]
+		}
+	}()
+
+	// Stream stderr to the stderrChannel
+	go func() {
+		defer close(stderrChannel)
+		for {
+			buf := make([]byte, 2048)
+			bytesRead, err := stderr.Read(buf)
+			if err != nil {
+				if err != io.EOF {
+					log.Fatal(err)
```

Comment:
Same goes here

Reply:
Looking to update and resolve this. I completely agree we should avoid this type of logging.